src/main/java/org/mockito/internal/util/eventbus/Subscriber.java

```diff
@@ -0,0 +1,78 @@
+package org.mockito.internal.util.eventbus;
+
+import static java.lang.System.identityHashCode;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import org.mockito.exceptions.base.MockitoException;
+
+public class Subscriber {
+
+    private final Object listenerInstance;
+    private final Method listenerMethod;
+    private final Class<?> eventType;
+
+    public Subscriber(Object listenerInstance, Method listenerMethod) {
+        this.listenerInstance = listenerInstance;
+        this.listenerMethod = listenerMethod;
+        this.eventType = listenerMethod.getParameterTypes()[0];
+    }
+
+    public final void tryHandleEvent(Object event) {
+        if (canHandle(event)) {
+            invokeListenerMethod(event);
+        }
+    }
+
+    protected void invokeListenerMethod(Object event) {
```

Comment:
Yes it is hacky, the error handling is also not final. In the current state the whole is a prototyp to see if it works at all. 

The idea is to allow methods and the declaring class to be non public. If the accessability is not changed we can't call public methods in private inner classes.